broker/src/main/java/org/apache/rocketmq/broker/out/BrokerOuterAPI.java

```diff
@@ -123,7 +124,7 @@ public void updateNameServerAddressList(final String addrs) {
         final int timeoutMills,
         final boolean compressed) {
 
-        final List<RegisterBrokerResult> registerBrokerResultList = Lists.newArrayList();
+        final List<RegisterBrokerResult> registerBrokerResultList = Collections.synchronizedList(Lists.newArrayList());
```

Comment:
I don't understand why we need a synchronized List here? Can u write a test case or sth to prove it?

Reply:
registerBrokerResultList is not threa-safe in the code:
for (final String namesrvAddr : nameServerAddressList) {
                brokerOuterExecutor.execute(new Runnable() {
                    @Override
                    public void run() {
                        try {
                            RegisterBrokerResult result = registerBroker(namesrvAddr,oneway, timeoutMills,requestHeader,body);
                            if (result != null) {
                                registerBrokerResultList.add(result);
                            }

                            log.info("register broker[{}]to name server {} OK", brokerId, namesrvAddr);
                        } catch (Exception e) {
                            log.warn("registerBroker Exception, {}", namesrvAddr, e);
                        } finally {
                            countDownLatch.countDown();
                        }
                    }
                });
            }